PowerValidatedSolutions.psm1

```diff
@@ -12939,6 +12939,127 @@ Function Test-vROPsAdapterStatusByType {
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that vRealize Operations Manager has been deployed in VMware Cloud Foundation aware mode and retrieves its details
+        - Validates that network connectivity and authentication is possible to vRealize Operations Manager
+        - Validates that vRealize Automation has been deployed in VMware Cloud Foundation aware mode and retrives its details
+        - Validates that network connectivity is possible to vRealize Automation
+        - Validates that the vRealize Automation adapter exists and configured in vRealize Operations Manager
+        - Validates that the credential name of vRealize Automation adapter does not already exist in vRealize Operations Manager
+        - Validates that the given credential is valid and updates vRealize Automation adapter in vRealize Operations Manager 
+        - Verifies the vRealize Automation adapter status in vRealize Operations Manager after updating the credential
+
+        .EXAMPLE
+        Update-vROPSvRAAdapterCredential -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -vRACredName "vRealize Automation Credentials" -vRAcredUserName svc-vrops-vra@sfo.rainpole.io -vRACredPass VMw@re1! 
+        This example update the credential of vRealize Automation adapter with name "vRealize Automation Credentials" in vRealize Operations Manager
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vRACredName,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vRAcredUserName,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vRACredPass
+    )
+
+    Try {
+        if (Test-VCFConnection -server $server) {
+            if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+                if (($vcfVropsDetails = Get-vROPsServerDetail -fqdn $server -username $user -password $pass)) {
+                    if (Test-vROPSConnection -server $vcfVropsDetails.loadBalancerFqdn) {   
+                        if (Test-vROPSAuthentication -server $vcfVropsDetails.loadBalancerFqdn -user $vcfVropsDetails.adminUser -pass $vcfVropsDetails.adminPass) {
+                            if (($vcfVraDetails = Get-vRAServerDetail -fqdn $server -username $user -password $pass)) {
+                                if (Test-vRAConnection -server $vcfVraDetails.loadBalancerFqdn) {  
+                                    if (Get-vROPSSolution | Where-Object { $_.adapterKindKeys -eq "CASAdapter" }) {
+                                        if ((Get-vROPSAdapter | Where-Object { $_.resourceKey.adapterKindKey -eq "CASAdapter" })) {
```

Comment:
Instead of adding `-eq "CASAdapter"` here should we instead be passing in `-eq $adapterKind` where it's pulled from a default parameter?